core/src/replay_stage.rs

```diff
@@ -3509,6 +3509,13 @@ impl ReplayStage {
         };
 
         if let Some((bank, switch_fork_decision)) = selected_fork {
+            let check_slot = if switch_fork_decision.can_vote() {
+                bank.slot()
+            } else {
+                // If we can't vote make the checks on the slot we wanted to vote on
+                // This is purely for logging/metrics, since we can't vote anyway
+                heaviest_bank.slot()
```

Comment:
This should be purely cosmetic. Will only fix the logging / metrics for `HeaviestForkFailures`